pkg/resource/key/key_rotation_hooks.go

```diff
@@ -0,0 +1,82 @@
+package key
+
+import (
+	"context"
+
+	ackrtlog "github.com/aws-controllers-k8s/runtime/pkg/runtime/log"
+	svcsdk "github.com/aws/aws-sdk-go/service/kms"
+)
+
+func (rm *resourceManager) updateKeyRotation(ctx context.Context, r *resource) (err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.updateKeyRotation")
+
+	defer func() {
+		exit(err)
+	}()
+
+	if r.ko.Spec.EnableKeyRotation == nil {
+		return nil
+	}
```

Comment:
Maybe we should disable the key rotation if a user deletes the `EnableKeyRotation` field?